### DISCUSSION TITLE
ALIAS in Form is not working

### DISCUSSION THREAD

**Discussion by harol97** [fastapi/fastapi#10024] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [x] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import FastAPI, File, Form, UploadFile

app = FastAPI()

@app.post("/testing-form")
def testing_form(id:Annotated[int ,Form(alias="myAlias")], file_custom:Annotated[UploadFile, File(alias="myFileAlias")]):
    print("Id >>", id)
    print("Filename >>", file_custom.filename)
    return {"status":"ok"}
```

### Description

I was working in a project when I started to have issues with Form() and Alias , after that create a new project

![Screenshot from 2023-08-06 21-44-34](https://github.com/tiangolo/fastapi/assets/49042862/0680167e-bcd8-4f4e-aa64-35acf65d38ed)

and I was testing alias in Form and can't see alias neither in swagger nor in redocs and it is not working. Also, i used postman and was not working.
![Screenshot from 2023-08-06 21-27-20](https://github.com/tiangolo/fastapi/assets/49042862/5e6cd246-c6e3-492e-9316-4c5fa4daf94f)

I started digging into this issue and compared the way "Form" uses the "alias" parameter vs. "BaseModel", I noticed that in the end they both use the FieldInfo class, but get it in a different way, so I extracted a piece of code that uses pydantic and put it inside the "if PYDANTIC_V2" in the Body class located at the FastApi params.py file and it worked correctly. I was hoping to fix it

![edit_fastapi_Body](https://github.com/tiangolo/fastapi/assets/49042862/01adadd7-d961-4171-88de-74a25b3344fc)

Am I missing something? or  is it incorrect?

Thank you in advance.
Best Regards.

### Operating System

Linux

### FastAPI Version

0.101.0

### Pydantic Version

2.1.1

### Python Version

3.11.3

**Comment by rifatrakib** [comment]
Is this getting fixed or what?

**Reply by harol97** [reply]
check [#10319 ](https://github.com/tiangolo/fastapi/pull/10319)

**Comment by Kludex** [comment, marked as answer]
- This can be tracked here: https://github.com/tiangolo/fastapi/issues/10286

For the time being, you can use `validation_alias`.

**Reply by harol97** [reply]
When I use only validation_alias I can see alias in Swagger, but the validation is not working well.
For the moment I'm using both alias and validation_alias, It works.
